src/iuse.cpp

```diff
@@ -2747,160 +2747,572 @@ int iuse::sew(player *p, item *it, bool, point)
+                   itm.made_of( "fur" ) ||
+                   itm.made_of( "nomex" ) ||
+                   itm.made_of( "plastic" ) ||
+                   itm.made_of( "wool" );
+        } );
+        item *mod = &(p->i_at(pos));
+        if (mod == NULL || mod->is_null()) {
+            p->add_msg_if_player(m_info, _("You do not have that item!"));
+            return 0;
+        };
+                itype_id repair_item = "none";
+        std::vector<std::string> plurals;
+        std::vector<itype_id> repair_items;
+        std::string plural = "";
+        //translation note: add <plural> tag to keep them unique
+        if (mod->made_of("cotton")) {
+            repair_items.push_back("rag");
+            plurals.push_back(rm_prefix(_("<plural>rags")));
+        }
+        if (mod->made_of("leather")) {
+            repair_items.push_back("leather");
+            plurals.push_back(rm_prefix(_("<plural>leather")));
+        }
+        if (mod->made_of("fur")) {
+            repair_items.push_back("fur");
+            plurals.push_back(rm_prefix(_("<plural>fur")));
+        }
+        if (mod->made_of("nomex")) {
+            repair_items.push_back("nomex");
+            plurals.push_back(rm_prefix(_("<plural>nomex")));
+        }
+        if (mod->made_of("plastic")) {
+            repair_items.push_back("plastic_chunk");
+            plurals.push_back(rm_prefix(_("<plural>plastic")));
+        }
+        if (mod->made_of("wool")) {
+            repair_items.push_back("felt_patch");
+            plurals.push_back(rm_prefix(_("<plural>wool")));
+        }
+        if (repair_items.empty()) {
+            p->add_msg_if_player(m_info, _("Your %s is not made of fabric, leather, fur, wool or plastic."),
+                                 mod->tname().c_str());
+            return 0;
         }
-        return 0;
-    }
+        if( mod == it
+                || std::find(repair_items.begin(), repair_items.end(), mod->typeId()) != repair_items.end()) {
+            p->add_msg_if_player(m_info, _("This can be used to repair other items, not itself."));
+            return 0;
+        };
+        if(((mod->item_tags.count("furred")) && (mod->item_tags.count("pocketed"))) ||
+           ((mod->item_tags.count("furred")) && (mod->item_tags.count("leather_padded"))) ||
+           ((mod->item_tags.count("pocketed")) && (mod->item_tags.count("leather_padded"))) ||
+           ((mod->item_tags.count("furred")) && (mod->item_tags.count("kevlar_padded"))) ||
+           ((mod->item_tags.count("kevlar_padded")) && (mod->item_tags.count("pocketed"))) ||
+           ((mod->item_tags.count("leather_padded")) && (mod->item_tags.count("kevlar_padded")))) { //This is a mess. Have to block every combo of 2. Probably a better way.
+            p->add_msg_if_player(m_info,_("You can't modify this more than twice."));
+            return 0;
+        };
+        int choice = menu(true, _("How do you want to modify it?"), _("Add extra straps and pockets"),
```

Comment:
If the tailoring kit was somehow separate from the sewing kit, that might be a possibility--I'm not sure what sort of sewing gear would permit adding linings and pockets yet *not* permit repairing a rip, but whatever.

But the tailoring kit contains a sewing kit, by its recipe, so I'm hard-pressed to understand exactly how one loses the repair functionality.  Further, the tailoring kit doesn't specify that it is somehow not suitable for repairs.  Folks may have used their sewing kit to make the tailor kit, and would not be pleased to learn that now they're expected to survivalcraft or scrounge another kit to regain their ability to repair.

Rivet's earlier point remains sound:
>No losing basic functionality in the process of upgrading.